src/ipc/ipc-zephyr.c

```diff
@@ -159,8 +235,8 @@ static int ipc_device_resume_handler(const struct device *dev, void *arg)
 	ipc->task_mask = 0;
 	ipc->pm_prepare_D3 = false;
 
-	/* attach handlers */
-	intel_adsp_ipc_set_message_handler(INTEL_ADSP_IPC_HOST_DEV, message_handler, ipc);
+	/* initialize IPC endpoint */
+	ipc_ept_init(ipc);
```

Comment:
in fact, can we not just use Zephyr functions directly here? Would make reading easier by saving a level of indirection

Reply:
This function also sets a pointer inside the private data to the IPC struct. We can expand both in the two places calling `ipc_ept_init()`. It's that if we ever have to change the init sequence, we don't have to do it at two places... and to avoid changing one and not the other.